Mangle libLLVM-VER.so to libLLVM-mesa-VER.so until LLVM gets proper SOVER
linking.

--- cmake/modules/AddLLVM.cmake.orig	2022-01-20 21:31:59 UTC
+++ cmake/modules/AddLLVM.cmake
@@ -610,7 +610,7 @@ function(llvm_add_library name)
     if(NOT APPLE AND ARG_SONAME)
       get_target_property(output_name ${name} OUTPUT_NAME)
       if(${output_name} STREQUAL "output_name-NOTFOUND")
-        set(output_name ${name})
+        set(output_name ${name}-mesa)
       endif()
       set(library_name ${output_name}-${LLVM_VERSION_MAJOR}${LLVM_VERSION_SUFFIX})
       set(api_name ${output_name}-${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH}${LLVM_VERSION_SUFFIX})
